test/corpus/expressions.txt

```diff
@@ -1001,6 +1001,195 @@ class Test {
               (expression_statement
                 (update_expression
                   (identifier))))))))
+
+======================================
+Switch statements & pattern matching
+======================================
+
+public class Switches {
+    static void patterns() {
+        Rectangle r = new Rectangle(new ColoredPoint(c1), new ColoredPoint(c2));
+        if (obj instanceof Point(int x)) {
+        }
+        if (r instanceof Rectangle(ColoredPoint ul)) {
+        }
+        if (r instanceof Rectangle(ColoredPoint(Point p, Color c),
+                  ColoredPoint lr)) {
+        }
+        if (r instanceof Rectangle(ColoredPoint(Point(var x, var y), var c),
+                    var lr)) {
+        }
+        switch (s) {
+            case Rectangle r: return 2;
+        }
+        switch (pair) {
+          case MyPair(var f, var s) -> 2;
+        }
+        if (bbs instanceof Box<String>(Box(var s))) {
+        }
+        return switch (s) {
+            case rectangle r -> 2;
+        };
+    }
+}
+
+---
+
+(program
```

Comment:
this is still 60Lines of expected output ... still too long. I would split the tests further, to better isolate the new supported features.